### PR TITLE
Fix a bug that produces an incorrect JSON for UDF return types.

### LINKED ISSUE #112: Array/Map return types for UDF do not work correctly.
I have a scenario where for structured streaming input and for each event/row i have to write a custom logic/function which can return multiple rows.

looks like for return type UDF only supports basic type and not list/array.

Any workaround for this?

for sample my UDF is something like below, so that i can explode to create multiple rows.
`
         Func<Column, Column> ToUpperList = Udf<string, string[]>((arg) =>
            {
                var ret = new string[] { arg, arg.ToUpper()};
                return ret;
            });

            var query = inStream
                .Select(Explode( ToUpperList(Col("InputEventColumn"))))
                .WriteStream()
                .OutputMode("append")
                .Format("console")
                .Start();`

### PR DESCRIPTION
**Pull request by imback82**
Fixes #112.

### CODE REVIEW

File: src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs [hunk 1]
```diff
@@ -119,16 +119,38 @@ public sealed class ShortType : IntegralType
     }
 
     /// <summary>
-    /// Represents a decimal type (not implemented).
+    /// Represents a decimal type.
     /// </summary>
     public sealed class DecimalType : FractionalType
     {
+        internal static Regex s_fixedDecimal =
+            new Regex(@"decimal\s*\((\d+),\s*(\d+)\)", RegexOptions.Compiled);
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs]: Are the \s* required ?

PySpark's spark/python/pyspark/sql/types.py:
```python
    def jsonValue(self):
        return "decimal(%d,%d)" % (self.precision, self.scale)
```

And the below:
```csharp
        public override string SimpleString => $"decimal({_precision},{_scale})";
```

doesn't contain spaces.

Comment (imback82) [reply src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs]: Yes. The true source should be https://github.com/apache/spark/blob/1e87694f2bc0dea3eaaef936dc8552b3ba421539/sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala#L117

Comment (suhsteve) [reply src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs]: Let's add the missing \s* after the (\d+) capture groups.

File: src/csharp/Microsoft.Spark/Utils/UdfUtils.cs [hunk 2]
```diff
@@ -49,47 +82,50 @@ internal enum PythonEvalType
...
-            // See spark.sql.datatypes.datatype.scala for reference.
-            var types = new Type[] { type };
-            var returnTypeFormat = "{0}";
-
-            if (type.IsArray)
+            if (s_returnTypes.TryGetValue(type, out string value))
             {
-                types[0] = type.GetElementType();
-                returnTypeFormat = "array<{0}>";
+                return $"\"{value}\"";
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/Utils/UdfUtils.cs]:
```suggestion
                return $@"""{value}""";
```

File: src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs (hunk 1 above)
Comment (imback82) [reply src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs]: Sure. Let me change that.

File: src/csharp/Microsoft.Spark/Utils/UdfUtils.cs (hunk 2 above)
Comment (imback82) [reply src/csharp/Microsoft.Spark/Utils/UdfUtils.cs]: fixed.

**Comment by suhsteve** [comment]
LGTM

**Pull request merged**